### PR TITLE
fix(Scripts/Raids): Some corrections to Broodlord Supression Room:

### LINKED ISSUE #10768: [BWL] Supression Room traps issues
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [x] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59
- [X] 60

### Current Behaviour

1. The Suppression debuff dont hit players. Runned thro all room both floor,  got hitted once. 
2. All suppressions goes down when broodlord dies. 
3. Traps that are disarmed you cannot see.

### Expected Blizzlike Behaviour

1.  with all suppression up there is 100% chance to be slowed when running pass.
2. When Broodlord dies,  all supressions that was down when he died stays down, but the rest should be up and needed to go back to press down if you want to do that.
3. Traps should still be visible in their disarmed state
### Source

Image showing:  1. Supression is up.  2. Broodlord has died 1 second before pic was taken,  the door has started to open as proof.

image

### Steps to reproduce the problem

Engage Broodlord  (3rd boss in BWL)

### Extra Notes

original: https://github.com/chromiecraft/chromiecraft/issues/3087

###Server

ChromieCraft PTR

### AC rev. hash/commit

rev. 982be6de0081

### Operating system

Ubuntu 20.04

### Modules

mod-anticheat (master) 
mod-autobalance (master)
mod-cfbg (master)
mod-chromiexp (master)
mod-chromiecraft (master)
mod-desertion-warnings (master)
mod-duel-reset (master)
mod-eluna-lua-engine (master)
mod-ip-tracker (master)
mod-low-level-arena (master)
mod-multi-client-check (master)
mod-progression-system (master)
mod-pvp-titles (master)
mod-pvpstats-announcer (master)
mod-queue-list-cache (master) 
mod-rdf-expansion (master) 
mod-save-inventory (master)
mod-server-auto-shutdown (master)
mod-spellsystem-extended (master)

### Customizations

Acore_eventScripts (master)
Acore_SendAndBind (master)
Acore_TempAnnouncements (master)
Acore_ZoneDebuff (master)

### PR DESCRIPTION
**Pull request by ghost**
Supression traps should always hit players.
Supression traps should not be disarmed after Broodlord defeat.
Supression traps should be visible after disarm.
Fixes #10768

## Issues Addressed:

- Closes #10768
- Closes https://github.com/chromiecraft/chromiecraft/issues/3087

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

`.gm on`
`.go xyz  -7579.5917 -980.4238 442 469`
Try to be hit by trap or disarm it.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Spells/SpellEffects.cpp [hunk 1]
```diff
@@ -2137,8 +2137,10 @@ void Spell::EffectOpenLock(SpellEffIndex effIndex)
         else if (m_spellInfo->Id == 1842 && gameObjTarget->GetGOInfo()->type == GAMEOBJECT_TYPE_TRAP)
         {
             gameObjTarget->SetLootState(GO_JUST_DEACTIVATED);
-            if (!gameObjTarget->GetOwner()) // pussywizard
+            if (gameObjTarget->getLootState() == GO_JUST_DEACTIVATED && !gameObjTarget->GetOwner()) // pussywizard
```
Comment (acidmanifesto) [inline src/server/game/Spells/SpellEffects.cpp:2140]: pussywizard? is this a partial revert? or from another project?

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: src/server/game/Spells/SpellEffects.cpp (hunk 1 above)
Comment (Nefertumm) [reply src/server/game/Spells/SpellEffects.cpp:2140]: I think the comment was there all the time

Comment (acidmanifesto) [reply src/server/game/Spells/SpellEffects.cpp:2140]: Ah gotcha. I must of overlooked it entirely.

**Comment by Lynethris** [comment]
Tested and 2/3 work: 

- [X] Supression traps always hit the player
- [X] Supression traps should be visible after disarm.
- [] Supression traps should not be disarm after Broodlord defeat

I can also certified : 
- [X] Traps can be only disarm by a rogue

Images:
<details>

WoWScrnShot_022722_085812

WoWScrnShot_022722_083438

WoWScrnShot_022722_085532

</details>

**Pull request merged**